symengine/ruby/ext/symengine/ruby_basic.c

```diff
@@ -66,7 +66,7 @@ VALUE cbasic_eq(VALUE self, VALUE operand2) {
     basic_struct *this, *cbasic_operand2;
 
     Data_Get_Struct(self, basic_struct, this);
-    Data_Get_Struct(operand2, basic_struct, cbasic_operand2);
+    sympify(operand2, &cbasic_operand2);
```

Comment:
After the commit [`ae56d65`](https://github.com/abinashmeher999/symengine/commit/ae56d65e3238339c1c9153c67989435670eb1d49), the error message has changed. Now it says that something is being freed twice, while earlier it said that I was not freeing some.
```
*** Error in `ruby /home/abinashmeher999/.rvm/gems/ruby-2.2.0/bin/rspec spec': double free or corruption (fasttop): 0x00000000026ef3e0 ***
Aborted (core dumped)
```
What am I missing here?